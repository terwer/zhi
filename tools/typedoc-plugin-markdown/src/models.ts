import { ReflectionKind, TypeDocOptionMap } from 'typedoc';

/**
 * Defines outputFileStrategy options
 * @enum
 */
export const OutputFileStrategy = {
  Modules: 'modules',
  Members: 'members',
} as const;

export type OutputFileStrategy =
  (typeof OutputFileStrategy)[keyof typeof OutputFileStrategy];

/**
 * Defines format style options
 * @enum
 */
export const FormatStyle = {
  List: 'list',
  Table: 'table',
} as const;

export type FormatStyle = (typeof FormatStyle)[keyof typeof FormatStyle];

/**
 * Defines format style options
 * @enum
 */
export const AnchorFormat = {
  Lowercase: 'lowercase',
  Slug: 'slug',
  None: 'none',
} as const;

export type AnchorFormat = (typeof AnchorFormat)[keyof typeof AnchorFormat];

export interface TypedocPluginMarkdownOptions extends TypeDocOptionMap {
  anchorFormat: AnchorFormat;
  anchorPattern: string;
  baseUrl: string;
  entryDocument: string;
  enumMembersFormat: FormatStyle;
  excludeGroups: boolean;
  flattenOutputFiles: boolean;
  hideBreadcrumbs: boolean;
  hideInPageTOC: boolean;
  hideKindTag: boolean;
  hidePageHeader: boolean;
  hidePageTitle: boolean;
  hideHierarchy: boolean;
  includeFileNumberPrefixes: boolean;
  indentifiersAsCodeBlocks: boolean;
  indexPageTitle: string;
  namedAnchors: boolean;
  outputFileStrategy: OutputFileStrategy;
  propertiesFormat: FormatStyle;
  typeDeclarationFormat: FormatStyle;
}

export interface TemplateMapping {
  directory: string | null;
  template: any;
  kind: ReflectionKind;
}

export type Collapse = 'object' | 'function' | 'all' | 'none';

export type FrontmatterGlobals =
  | string
  | Record<string, string | boolean | number | null>;

export interface NavigationItem {
  title: string;
  url?: string;
  children?: NavigationItem[];
  isReadme?: boolean;
  isGroup?: boolean;
}
