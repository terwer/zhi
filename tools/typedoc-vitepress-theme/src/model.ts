export interface SidebarOptions {
  autoConfiguration: boolean;
  format: 'vitepress' | 'vuepress1' | 'vuepress2';
}
