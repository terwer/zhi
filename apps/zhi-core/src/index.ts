/*
 * Copyright (c) 2023, Terwer . All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Terwer designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Terwer in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Terwer, Shenzhen, Guangdong, China, youweics@163.com
 * or visit www.terwer.space if you need additional information or have any
 * questions.
 */

/**
 * @packageDocumentation
 * zhi-core 主题核心模块
 */

import Zhi from "./theme/zhi"
import DeviceDetection from "zhi-device-detection"

// 主题样式注入入口
import "./style/common/fonts/webfont.css"
import "./style/index.styl"

const loadTheme = async (): Promise<void> => {
  const zhi = new Zhi(DeviceDetection.getDevice())
  await zhi.init()
}

/**
 * 主题入口，由思源笔记自动触发，请勿主动调用
 */
;(async () => {
  await loadTheme()
})()

export default loadTheme
export { Zhi }
