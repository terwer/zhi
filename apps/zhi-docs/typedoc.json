{
  "plugin": [
    "typedoc-plugin-frontmatter",
    "@terwer/typedoc-plugin-markdown",
    "@terwer/typedoc-plugin-markdown"
  ],
  "entryPoints": [
    "../../libs/zhi-device/src/index.ts",
    "../../libs/zhi-env/src/index.ts",
    "../../libs/zhi-log/src/index.ts",
    "../../libs/zhi-common/src/index.ts",
    "../../libs/zhi-blog-api/src/index.ts",
    "../../libs/zhi-fetch-middleware/src/index.ts",
    "../../libs/zhi-xmlrpc-middleware/src/index.ts"
  ],
  "tsconfig": "./tsconfig.json",
  "frontmatterTags": ["author", "description", "outline"],
  "frontmatterGlobals": {
    "outline": "deep"
  }
}