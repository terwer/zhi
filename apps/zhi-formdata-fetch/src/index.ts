import FormdataFetch from "./lib/formdataFetch"
import fetch, { Blob, FormData, Headers } from "node-fetch"

export { FormdataFetch }
export { fetch }
export { Blob, FormData, Headers }
