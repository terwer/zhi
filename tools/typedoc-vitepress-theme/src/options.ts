export const DEFAULT_SIDEBAR_OPTIONS = {
  autoConfiguration: true,
  format: 'vitepress',
};
