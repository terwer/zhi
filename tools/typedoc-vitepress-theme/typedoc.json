{
  "entryPoints": [
    "../../../stubs/src/classes.ts",
    "../../stubs/src/comments.ts"
  ],
  "tsconfig": "../../../stubs/tsconfig.json",
  "plugin": ["typedoc-plugin-markdown", "../dist"],
  "sidebar": { "version": "v1" }
}
