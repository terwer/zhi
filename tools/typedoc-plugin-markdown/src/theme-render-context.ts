import * as path from 'path';
import { Options, Reflection } from 'typedoc';
import { TypedocPluginMarkdownOptions } from './models';
import { partials, templates } from './resources/resources';
import { URL_PREFIX } from './support/constants';
import { MarkdownTheme } from './theme';

export class MarkdownThemeRenderContext {
  public readmeFile = 'README.md';
  public indexFile = 'index.md';
  public modulesFile = 'modules.md';
  public exportsFile = 'exports.md';

  private _activeLocation = '.';

  set activeLocation(activeLocation: string) {
    this._activeLocation = activeLocation;
  }

  get activeLocation() {
    return this._activeLocation;
  }

  constructor(public theme: MarkdownTheme, public options: Options) {}

  getOption<K extends keyof TypedocPluginMarkdownOptions>(name: K) {
    return this.options.getValue(name) as TypedocPluginMarkdownOptions[K];
  }

  getRelativeUrl(url: string | undefined) {
    if (!url) {
      return null;
    }
    if (URL_PREFIX.test(url)) {
      return url;
    } else {
      if (this.getOption('baseUrl')) {
        return this.getOption('baseUrl') + url.replace(/\\/g, '/');
      }

      const relative = path.relative(
        path.dirname(this.activeLocation),
        path.dirname(url),
      );

      return this.parseUrl(
        path.join(relative, path.basename(url)).replace(/\\/g, '/'),
      );
    }
  }

  urlTo(reflection: Reflection) {
    return this.relativeURL(reflection.url);
  }

  parseUrl(url: string) {
    return encodeURI(url);
  }

  relativeURL(url: string | undefined) {
    return this.getRelativeUrl(url);
  }

  templates = templates(this);
  partials = partials(this);
}
