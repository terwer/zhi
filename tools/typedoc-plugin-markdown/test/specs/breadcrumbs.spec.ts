import { ProjectReflection, Reflection } from 'typedoc';

import { MarkdownThemeRenderContext } from '../../src/theme-render-context';

describe(`Breadcrumbs:`, () => {
  let moduleReflection: Reflection;
  let classReflection: Reflection;

  describe(`(with readme)`, () => {
    let project: ProjectReflection;
    let context: MarkdownThemeRenderContext;
    beforeAll(async () => {
      ({ project, context } = await global.bootstrap(['breadcrumbs.ts']));
      moduleReflection = (project.children as any)[0];
      classReflection = project.getChildByName('Breadcrumbs') as Reflection;
      context.activeLocation = 'breadcrumbs.md';
    });

    test(`should compile entryPoint (globals) breadcrumbs'`, () => {
      expect(
        context.partials.breadcrumbs({
          project: project,
          model: project,
          url: 'globals.md',
        } as any),
      ).toMatchSnapshot();
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        context.partials.breadcrumbs({
          project: project,
          model: moduleReflection,
          url: moduleReflection.url,
        } as any),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        context.partials.breadcrumbs({
          project: project,
          model: classReflection,
          url: classReflection.url,
        } as any),
      ).toMatchSnapshot();
    });
  });
  describe(`(without readme)`, () => {
    let project: ProjectReflection;
    let context: MarkdownThemeRenderContext;
    beforeAll(async () => {
      ({ project, context } = await global.bootstrap(['breadcrumbs.ts'], {
        options: { readme: 'none' },
      }));
      moduleReflection = (project.children as any)[0];
      classReflection = project.getChildByName('Breadcrumbs') as Reflection;
      context.activeLocation = 'breadcrumbs.md';
    });

    test(`should compile module breadcrumbs'`, () => {
      expect(
        context.partials.breadcrumbs({
          project: project,
          model: moduleReflection,
          url: moduleReflection.url,
        } as any),
      ).toMatchSnapshot();
    });
    test(`should compile class breadcrumbs'`, () => {
      expect(
        context.partials.breadcrumbs({
          project: project,
          model: classReflection,
          url: classReflection.url,
        } as any),
      ).toMatchSnapshot();
    });
  });
});
