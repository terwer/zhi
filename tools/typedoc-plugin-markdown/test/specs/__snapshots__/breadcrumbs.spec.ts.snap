// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs: (with readme) should compile class breadcrumbs' 1`] = `"[Exports](exports.md) > Breadcrumbs"`;

exports[`Breadcrumbs: (with readme) should compile entryPoint (globals) breadcrumbs' 1`] = `"[Exports](exports.md) > typedoc-plugin-markdown"`;

exports[`Breadcrumbs: (with readme) should compile module breadcrumbs' 1`] = `"[Exports](exports.md) > Breadcrumbs"`;

exports[`Breadcrumbs: (without readme) should compile class breadcrumbs' 1`] = `"[Exports](README.md) > Breadcrumbs"`;

exports[`Breadcrumbs: (without readme) should compile module breadcrumbs' 1`] = `"[Exports](README.md) > Breadcrumbs"`;
